Modules/socketmodule.c

```diff
@@ -1694,10 +1696,13 @@ getsockaddrarg(PySocketSockObject *s, PyObject *args,
                 goto unix_out;
             }
             addr->sun_path[path.len] = 0;
+
+            /* including the trailing NUL */
+            *len_ret = path.len + offsetof(struct sockaddr_un, sun_path) + 1;
         }
         addr->sun_family = s->sock_family;
         memcpy(addr->sun_path, path.buf, path.len);
-        *len_ret = path.len + offsetof(struct sockaddr_un, sun_path);
+        
```

Comment:
perhaps add this assertion here:

```
#ifdef __linux__
        if (path.len > 0 && *(const char *)path.buf != 0)
#endif
        {
            /* Check for https://bugs.python.org/issue44493 */
            assert(((const char *)addr)[*len_ret-1] == 0);
        }
```

gross, but on our debug builds that should catch the problem happening. Our existing test_socket will crash on that assert without your fix.

Reply:
What if some platform didn't put the `sun_path` as the last field of `struct sockaddr_un`? Maybe we should only assert the last byte of `sun_path`.

Reply:
I don't believe it is possible for `sockaddr_un` to have fields after `sun_path` and even be compatible with length based APIs none of which would consider additional fields?  For an additional field to be used it would need to be a non standard API extension that requires `sun_path` to have a special value and the length to go beyond the end of `offsetof(sockaddr_un, sun_path) + sizeof(sockaddr_un.sun_path)`

regardless unlikely things to think about like that suggest just not bothering with such an assert.

I wanted to write it in such a way that was not simply repeating the len_buf calculation.  it's more complicated that I'd like as is, so we may as well not add it. :)